pkg/util/winutil/closedsourceconsent.go

```diff
@@ -0,0 +1,115 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+// +build windows
+
+package winutil
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"golang.org/x/sys/windows/registry"
+)
+
+const (
+	AgentRegistryKey    = `SOFTWARE\DataDog\Datadog Agent`
+	ClosedSourceKeyName = `AllowClosedSource`
+	ClosedSourceAllowed = 1
+	ClosedSourceDenied  = 0
+)
+
+// This function retrieves the current consent value from the Winows registry
+func GetClosedSourceConsent() (consentVal uint64, err error) {
+
+	// try to open the Datadog agent reg key
+	regKey, err := registry.OpenKey(registry.LOCAL_MACHINE, AgentRegistryKey, registry.QUERY_VALUE)
+	if err != nil { // couldn't open
+		log.Warnf("unable to open registry key %s: %v", AgentRegistryKey, err)
+		return // consentVal will be 0 (denied)
+	}
+	defer regKey.Close()
+
+	// get the value for AllowClosedSource
+	consentVal, _, err = regKey.GetIntegerValue(ClosedSourceKeyName)
+	if err != nil {
+		log.Warnf("unable to get value for %s: %v", ClosedSourceKeyName, err)
+		return // consentVal will be 0 (denied)
+	}
+	return
+}
+
+// Determine if closed source is allowed or denied on the host
+func IsClosedSourceAllowed() (allowed bool, err error) {
+	val, err := GetClosedSourceConsent()
+	if err != nil {
+		return
+	}
+	return (val == ClosedSourceAllowed), err
+}
+
+// Allow closed source, if already allowed, does nothing
+func AllowClosedSource() error {
```

Comment:
Since `AllowClosedSource` is the name of the key, and the config option, suggest
`SetAllowClosedSource` to make it clear this is changing the config vs. getting the option.

Could also combine with `DenyClosedSource()` to end up with
`func SetAllowClosedSource(onoff bool) err`